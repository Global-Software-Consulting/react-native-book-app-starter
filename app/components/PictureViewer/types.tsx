import { ImageSourcePropType } from 'react-native';

export interface Props {
    isVisible: boolean;
    imageSource: ImageSourcePropType;
    onPress: () => void;
}
