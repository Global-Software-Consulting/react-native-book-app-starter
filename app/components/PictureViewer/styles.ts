import { StyleSheet } from 'react-native';
import { Dimensions } from 'react-native';
import { getPercentageWidth, getPercentageHeight } from 'utils/dimentionUtil';

export const useStyles = () => {
    const width = getPercentageWidth();
    const height = getPercentageHeight();
    const windowWidth = Dimensions.get('window').width;
    const windowHeight = Dimensions.get('window').height;
    const styles = StyleSheet.create({
        mainView:
            {  backgroundColor: 'white', alignSelf: 'center' },
        closeText:
            { color: 'red', alignSelf: 'center' },
        image:
            { width: windowHeight > windowWidth ? width('90') : height('80%'), height:windowHeight > windowWidth ? height('50%') : width('30%') },
        closeButton: {
            alignContent: 'center',
            alignItems: 'center',
            
        },
        text:
        {
            color: 'red',
            fontWeight: 'bold',
            marginBottom:20
        }
        
        
    });
    return styles;
};
