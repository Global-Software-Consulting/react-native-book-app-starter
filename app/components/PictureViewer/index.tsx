import React from 'react';
import { View, Image, StyleSheet, TouchableOpacity, Button } from 'react-native';
import Modal from 'react-native-modal';
import { Text } from 'react-native-paper';
import { useTranslation } from 'react-i18next';
import { Props } from './types';
import { useStyles } from './styles';
const PictureViewer: React.FC<Props> = (props) => {
    const { t } = useTranslation();
    const styles = useStyles();
    return (
        <Modal isVisible={props.isVisible} animationIn="slideInUp">
            <View style={styles.mainView}>
                <Image source={props.imageSource} style={styles.image} />
            </View>
            <TouchableOpacity style={styles.closeButton} onPress={props.onPress}>
                <Text style={styles.text}>{t('Close')}</Text>
            </TouchableOpacity>
        </Modal>
    );
};
export default PictureViewer;
