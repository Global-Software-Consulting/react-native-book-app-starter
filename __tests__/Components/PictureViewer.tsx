import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import PictureViewer from './../../app/components/PictureViewer/index'

describe('Component testing', () => {
    test('Snapshot', () => {
        const tree = render(<PictureViewer isVisible={true} imageSource={''} onPress={() => {}} />)
        expect(tree).toMatchSnapshot()
    })
    test('close action', () => {
        const { getByTestId } = render(
            <PictureViewer isVisible={true} imageSource={''} onPress={() => {}} />
        )
        // fireEvent.press(getByTestId('search'));
        fireEvent.press(getByTestId('close'))
    })
})
