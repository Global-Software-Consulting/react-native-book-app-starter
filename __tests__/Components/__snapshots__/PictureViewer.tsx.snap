// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Snapshot 1`] = `
<mockedDefaultExport
  animationIn="slideInUp"
  isVisible={true}
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
      }
    }
  >
    <Image
      source=""
      style={
        Object {
          "height": 667,
          "width": 675,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "opacity": 1,
      }
    }
    testID="close"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "red",
            "fontWeight": "bold",
            "marginBottom": 20,
          },
        ]
      }
    />
  </View>
</mockedDefaultExport>
`;
